tests/lib/tools/boot-state

```diff
@@ -0,0 +1,197 @@
+#!/bin/bash
+
+show_help() {
+    echo "usage: boot-state bootenv show [var]"
+    echo "       boot-state bootenv set <var> <value>"
+    echo "       boot-state bootenv unset <var>"
+    echo "       boot-state boot-path"
+    echo "       boot-state wait-core-post-boot"
+    echo ""
+    echo "Get information and manages the boot loader for the current system"
+    echo ""
+    echo "COMMANDS:"
+    echo "  bootenv show:         prints the whole bootenv or just the variable passed as parameter"
+    echo "  bootenv set:          sets the given var and value on boot configuration"
+    echo "  bootenv unset:        unsets the given var from boot configuration"
+    echo "  boot-path:            prints the boot path"
+    echo "  wait-core-post-boot:  waits until the snap_mode bootenv var is empty"
+}
+
+get_grub_editenv() {
+    if [ -e /boot/grub2/grubenv ]; then
+        command -v grub2-editenv
+    elif [ -e /boot/grub/grubenv ]; then
+        command -v grub-editenv
+    fi
```

Comment:
else - should it fails? Otherwise if it doesn't return a command, the error from run_grub_editenv will be very unclear.

Reply:
It is done intentionally, but I'll update to make it fail and catch the failure in a following pr
@stolowski thanks for the suggestion.